CMakePresets.json

```
@@ -67,6 +67,17 @@
       "inherits": [
         "base"
       ]
+    },
+    {
+      "name": "iwyu",
+      "displayName": "IWYU",
+      "description": "Fix includes with include what you use",
+      "cacheVariables": {
+        "ENABLE_IWYU": "ON"
+      },
+      "inherits": [
+        "base"
+      ]
```

I think it's safer to force CMake to always use clang with iwyu, so maybe create a minimal preset that can be re-used
```suggestion
        "clang"
      ]
```

and then at the top we can have
```json
    {
      "name": "clang",
      "environment": {
        "CC": "clang",
        "CXX": "clang++"
      },
      "hidden": true
    },
```